Shoko.Server/API/v2/Modules/PlexWebhook.cs

```diff
@@ -1,348 +1,643 @@
+        }
+
+        [NonAction]
+        public DateTime FromUnixTime(long unixTime)
+        {
+            return new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddSeconds(unixTime);
+        }
+
+        [Authorize]
+        [HttpGet("loginurl")]
+        public string GetLoginUrl() => CallPlexHelper(h => h.LoginUrl);
+
+        [Authorize]
+        [HttpGet("pin/authenticated")]
+        public bool IsAuthenticated() => CallPlexHelper(h => h.IsAuthenticated);
+
+        [Authorize]
+        [HttpGet("token/invalidate")]
+        public bool InvalidateToken() => CallPlexHelper(h => { h.InvalidateToken(); return true; });
+
+        [Authorize]
+        [HttpGet("sync")]
+        public ActionResult Sync()
+        {
+            Analytics.PostEvent("Plex", "SyncOne");
+
+            new CommandRequest_PlexSyncWatched(HttpContext.GetUser()).Save();
+            return APIStatus.OK();
+        }
+
+        [Authorize("admin")]
+        [HttpGet("sync/all")]
+        public ActionResult SyncAll()
+        {
+            ShokoServer.Instance.SyncPlex();
+            return APIStatus.OK();
+        }
+
+        [Authorize("admin")]
+        [HttpGet("sync/{id}")]
+        public ActionResult SyncForUser(int uid)
+        {
+            JMMUser user = HttpContext.GetUser();
+            ShokoServer.Instance.SyncPlex();
+            return APIStatus.OK();
+        }
+
+
+#if DEBUG
+        [Authorize]
+        [HttpGet("test/dir")]
+        public Directory[] GetDirectories() => CallPlexHelper(h => h.GetDirectories());
+
+        [Authorize]
+        [HttpGet("test/lib/{id}")]
+        public PlexLibrary[] GetShowsForDirectory(int id) => CallPlexHelper(h => ((SVR_Directory)h.GetDirectories().FirstOrDefault(d => d.Key == id))?.GetShows());
+#endif
+
+        [NonAction]
+        private T CallPlexHelper<T>(Func<PlexHelper, T> act)
+        {
+            JMMUser user = HttpContext.GetUser();
+            return act(PlexHelper.GetForUser(user));
+        }
+
+        #region Plex Event Models
+
+        /// <summary>
+        /// The plex web-hook event.
+        /// <br/>
+        /// See https://support.plex.tv/articles/115002267687-webhooks for some more details.
+        /// </summary>
+        #pragma warning disable 0649
+        public class PlexEvent
+        {
+            /// <summary>
+            /// The event name.
+            /// </summary>
+            [Required]
+            [DataMember(Name = "event")]
+            public string Event;
+
+            /// <summary>
+            /// If the user triggered the event.
+            /// <br/>
+            /// The event can be triggered by both the
+            /// <see cref="PlexEvent.Owner" /> and the
+            /// <see cref="PlexEvent.User" /> simultaneously.
+            /// </summary>
+            [DataMember(Name = "user")]
+            public bool User;
+
+            /// <summary>
+            /// If the server owner triggered the event.
+            /// </summary>

```

Comment:
the docblocks are pointless, we only needed the annotation.

Reply:
:crying_cat_face: